common/Subprocess.cc

```diff
@@ -53,35 +53,63 @@ class FileActions {
 };
 } // namespace
 
-// can't take string_views because we need char * not const char *
-optional<string> sorbet::Subprocess::spawn(string executable, vector<string> arguments) {
+// Spawns a new child process, pipes `stdinContents` into the new processes's stdin,
+// and returns the child's stdout and status code
+optional<sorbet::Subprocess::Result> sorbet::Subprocess::spawn(string executable, vector<string> arguments,
```

Comment:
This API is already in use (in `compiler/Linker/Linker.cc`).

The changes inside this method don't look to me to be entirely about performance improvements? It also looks like we're unconditionally piping some standard streams that we were not previously. Will that break the existing usage in Linker.cc?